Makefile

```diff
@@ -395,29 +402,27 @@ allexamples = $(examples) $(external_examples)
 search ?= t ex
 realsearch = $(search:%=%%)
 matched = $(foreach pattern,$(realsearch),$(filter $(OBJDIR)/$(pattern),$(tests) $(allexamples)))
-# Work around Nek examples not having normal targets
-matched_prereq = $(filter-out $(OBJDIR)/nek%,$(matched)) $(if $(findstring nek,$(matched)),prepnektests)
 
 # Test core libCEED
 test : $(matched:$(OBJDIR)/%=run-%)
 
-# run test target in parallel
+# Run test target in parallel
 tst : ;@$(MAKE) $(MFLAGS) V=$(V) test
 # CPU C tests only for backend %
 ctc-% : $(ctests);@$(foreach tst,$(ctests),$(tst) /cpu/$*;)
 
 prove : BACKENDS += $(TEST_BACKENDS)
-prove : $(matched_prereq)
+prove : $(matched)
 	$(info Testing backends: $(BACKENDS))
 	$(PROVE) $(PROVE_OPTS) --exec 'tests/tap.sh' $(matched:$(OBJDIR)/%=%)
-# run prove target in parallel
+# Run prove target in parallel
 prv : ;@$(MAKE) $(MFLAGS) V=$(V) prove
 
-prepnektests:
-	(export CC FC && cd examples && make prepnektests)
-
-prove-all :
-	+$(MAKE) prove realsearch=%
+prove-all-prep : lib
+	$(MAKE) -j1 examples
+prove-all : prove-all-prep
+	+$(MAKE) CC=$(CC) FC=$(FC) CEED_DIR=`pwd` NEK5K_DIR="$(abspath $(NEK5K_DIR))" \
+prove realsearch=%
```

Comment:
How about this? It runs in parallel (I believe) but builds in serial to avoid the Nek conflicts?

Reply:
Hmm, only the Nek part needs to be serial. This seems to serialize more than necessary.

Reply:
I think my latest commit only does the Nek examples in serial
```
prove-all-prep :
	+$(MAKE) NEK5K_DIR= examples
	$(MAKE) -j1 examples
```

Reply:
@jedbrown, I think I fixed this. Are we ok to merge?